### DISCUSSION TITLE
How do you guys deal with swap inside jails?

### DISCUSSION THREAD

**Discussion by qudiqudi** [Jip-Hop/jailmaker#172] [General]
Inside my Debain jails swap is always getting filled up. I don't see performance impacts because jailmaker is running on SSD storage, but it's always bugging me in some sort.

```
root@docker:~$ free -m
               total        used        free      shared  buff/cache   available
Mem:           64057       45169       17412         285        1475       17556
Swap:           4091        4073          18
```
Setting [swappiness](https://www.redhat.com/sysadmin/clear-swap-linux) to a low value or even 0 has no effect apparently in:
`cat /proc/sys/vm/swappiness`
As well as swapoff -a.

I think because proc/sys/* is bound to the base system, right?
So how do you handle that? Or do you just ignore it?

**Comment by Jip-Hop** [comment]
Could this be related?
https://github.com/Jip-Hop/jailmaker/discussions/155#discussioncomment-9441727

**Reply by qudiqudi** [reply]
No, I don't think so. I'm still on Cobia and the base system is rock solid, as I said, it's mainly an aesthetics thing, because memory swapping between ram and fast flash storage doesn't actually impact system performance. 
Just wondered if anyone is using some sort of script to kill the swap or anything similar?

**Reply by Jip-Hop** [reply]
What do you see on the host when you run `free -m`?

I see exactly the same output both inside and outside of the jail:

```
free -m
               total        used        free      shared  buff/cache   available
Mem:           64056       31222       32012         388        1920       32833
Swap:              0           0           0
```

If you have multiple jails, you could also try to see if the output varies between them. I suspect jails don't swap and it's a host-only thing.

**Reply by qudiqudi** [reply]
You're spot on on that, free -m shows identical output on the host and all jails.
And it confirms my assumption, swap is handled exclusively by the host and systemd is only punching through the sys settings - one way.

**Reply by qudiqudi** [reply]
I found this intresting read in the truenas forum. After reading that, I'm less worried about the swap always filled up to 100%

> On SSDs, swapping out anonymous pages and reclaiming file pages are essentially equivalent in terms of performance and latency. 

https://chrisdown.name/2018/01/02/in-defence-of-swap.html

**Comment by qudiqudi** [comment]
Swap is good thing. Don't worry about it. 

https://chrisdown.name/2018/01/02/in-defence-of-swap.html
